node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/util/ConfigServerHttpRequestExecutor.java

```diff
@@ -50,7 +50,12 @@
             Executors.newScheduledThreadPool(1, ThreadFactoryFactory.getDaemonThreadFactory("http-client-refresher"));
     private final List<URI> configServerHosts;
 
-    private SelfCloseableHttpClient client;
+    /**
+     * The client will be periodically re-created by clientRefresherScheduler if we provide keyStoreOptions
+     * or trustStoreOptions. This is needed because the key/trust stores are updated outside of node-admin,
```

Comment:
This comment doesn't say why 'volatile' is needed. I was thinking something like:
The 'client' reference must be volatile because it is set and read in different threads, and visibility of changes is only guaranteed for volatile variables.

Reply:
Added.